app/controllers/groups_controller.rb

```diff
@@ -311,7 +310,27 @@ def add_members
       end
     end
 
-    if (usernames = group.users.where(id: users.pluck(:id)).pluck(:username)).present?
+    emails = []
+    if params[:emails]
+      params[:emails].split(",").each do |email|
+        existing_user = User.find_by_email(email)
+        existing_user.present? ? users.push(existing_user) : emails.push(email)
+      end
+    end
+
+    if users.empty? && emails.empty?
+      raise Discourse::InvalidParameters.new(
+        'usernames or emails must be present'
+      )
+    end
+
+    if users.uniq.length != users.length
```

Comment:
I'm not sure we need to raise an error for this. You could pass `users.uniq` to your future query and it'll silently ignore any duplicates.